File: DashboardAdmin/src/app/services/review/review.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ReviewModel } from './review.model';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  private baseUrl = 'http://localhost:8888/api/reviews'; // change to gateway if needed

  constructor(private http: HttpClient) {}

  // 🔵 Get all reviews
  getAllReviews(): Observable<ReviewModel[]> {
    return this.http.get<ReviewModel[]>(this.baseUrl);
  }

  // 🔵 Get review by ID
  getReviewById(id: string): Observable<ReviewModel> {
    return this.http.get<ReviewModel>(`${this.baseUrl}/${id}`);
  }

  // 🔵 Get reviews by product ID
  getReviewsByProductId(productId: string): Observable<ReviewModel[]> {
    return this.http.get<ReviewModel[]>(`${this.baseUrl}/product/${productId}`);
  }

  // 🔵 Get reviews by user ID
  getReviewsByUserId(userId: string): Observable<ReviewModel[]> {
    return this.http.get<ReviewModel[]>(`${this.baseUrl}/user/${userId}`);
  }

  // 🟢 Create review
  createReview(review: ReviewModel): Observable<ReviewModel> {
    return this.http.post<ReviewModel>(this.baseUrl, review);
  }

  // 🟡 Update review
  updateReview(id: string, review: ReviewModel): Observable<ReviewModel> {
    return this.http.put<ReviewModel>(`${this.baseUrl}/${id}`, review);
  }

  // 🔴 Delete review
  deleteReview(id: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: backend/ReviewService/ReviewService/src/main/java/com/ReviewService/ReviewService/Service/Interface/ReviewService.java
package com.ReviewService.ReviewService.Service.Interface;




import java.util.List;
import java.util.Optional;

import com.ReviewService.ReviewService.DTO.ReviewDTO;

public interface ReviewService {
    ReviewDTO saveReview(ReviewDTO reviewDTO);
    ReviewDTO updateReview(String id, ReviewDTO reviewDTO);
    Optional<ReviewDTO> getReviewById(String id);
    List<ReviewDTO> getAllReviews();
    List<ReviewDTO> getReviewsByProductId(String productId);
    List<ReviewDTO> getReviewsByUserId(String userId);
    void deleteReview(String id);
    boolean hasUserReviewedProduct(String userId, String productId);
}
